File: source/WeddingManager.Services/ServiceService.cs
using System.Collections.Generic;
using WeddingManager.Core.Data;
using WeddingManager.Core.Repositories;
using WeddingManager.Core.Services;

namespace WeddingManager.Services
{
    public class ServiceService : IServiceService
    {
        public ServiceService(IServiceRepository serviceRepository)
        {
            _serviceRepository = serviceRepository;
        }

        public int CreateService(int customerId, Service service)
        {
            return _serviceRepository.CreateService(customerId, service);
        }

        public IEnumerable<Service> RetrieveServices(int customerId)
        {
            return _serviceRepository.RetrieveServices(customerId);
        }

        public Service RetrieveService(int companyId, int serviceId)
        {
            return _serviceRepository.RetrieveService(companyId, serviceId);
        }

        public void UpdateService(Service service)
        {
            _serviceRepository.UpdateService(service);
        }

        public void DeleteService(int serviceId)
        {
            _serviceRepository.DeleteService(serviceId);
        }

        public IEnumerable<Service> Search(int companyId, ServiceSearchCriteria searchCriteria)
        {
            return _serviceRepository.Search(companyId, searchCriteria);
        }

        private readonly IServiceRepository _serviceRepository;
    }
}

File: source/WeddingManager.Web/js/services/serviceService.js
(function(){
    angular.module('weddingManager')
    .service('serviceService', function(serviceFactory, appStateService){
        var output = {};
        var _services = [];
        var _searchResults = [];
        
        output.createService = function(customerId, onSuccess, onError){
            var service = {
                Description : "New service",
            };
            serviceFactory.createService(customerId, service, function(serviceId){
                service.Id = serviceId;
                onSuccess(service);
            }, onError);
        }
        
        output.refreshServices = function(customerId, onSuccess, onError){
            serviceFactory.retrieveServices(customerId, function(services){
                _services = services;
                onSuccess(services);
            }, onError);
        }
        
        output.getServices = function(){
            return _services;
        }
        
        output.retrieveService = function(companyId, serviceId, onSuccess, onError){
            serviceFactory.retrieveService(companyId, serviceId, onSuccess, onError);
        }
        
        output.updateService = function(service, onSuccess, onError){
            serviceFactory.updateService(service, onSuccess, onError);
        }
        
        output.deleteService = function(serviceId, onSuccess, onError){
            serviceFactory.deleteService(serviceId, function(){
                appStateService.setService(null);
                onSuccess();
            }, onError);
        }
        
        output.search = function(companyId, searchCriteria, onSuccess, onError){
            serviceFactory.search(companyId, searchCriteria, function(searchResults){
                _searchResults = searchResults;
                onSuccess(searchResults);
            }, onError);
        }
        
        output.getSearchResults = function(){
            return _searchResults;
        }
        
        return output;
    });
})();